libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -1278,7 +1280,11 @@ envFromPred env p = case p of
 -- | A version of `genFromSpecT` that simply errors if the generator fails
 genFromSpec :: forall fn a. (HasCallStack, HasSpec fn a) => Specification fn a -> Gen a
 genFromSpec spec = do
-  res <- strictGen $ explain1 "Calling genFromSpec" $ genFromSpecT spec
+  res <- strictGen $ explain1 "Calling genFromSpec" $ do
+    r <- genFromSpecT spec
+    unsafePerformIO $
+      r `seq`
+        pure (pure r) `catch` \(e :: SomeException) -> pure (fatalError (pure $ show e))
```

Comment:
Is it just calls to error that we are trying to catch, or something else?

Reply:
Calls to error yes

Reply:
This is a horrible piece of code that is totally incorrect:

1. It will catch all exceptions, including async exceptions. It should have been catching the `ErrorCall` exception instead of `SomeException`, if it needs to catch error calls.

2. Moreover this code will not even catch error calls because `seq` happens outside of `catch`:
```haskell
ghci> let x = undefined in x `seq` pure x `catch` \e -> putStrLn ("WTH " <> show (e :: SomeException))
*** Exception: Prelude.undefined
CallStack (from HasCallStack):
  undefined, called at <interactive>:8:9 in interactive:Ghci2
ghci> let x = undefined in (x `seq` pure x) `catch` \e -> putStrLn ("WTH ==============\n" <> show (e :: SomeException))
WTH ==============
Prelude.undefined
CallStack (from HasCallStack):
  undefined, called at <interactive>:10:9 in interactive:Ghci3
```
3. Moreover, using `unsafePerformIO` to catch an exception will lead to non-deterministic execution, which can break referential transparency. In other words it is never safe to catch exceptions in pure code with `unsafePerformIO`